humble build errors tf2 and rclcpp_components

When building for humble these build errors currently appear:

```sh
ros-humble-novatel-gps-driver-4.1.0/src/novatel_gps.cpp: In member function ‘void novatel_gps_driver::NovatelGps::GenerateImuMessages()’:
ros-humble-novatel-gps-driver-4.1.0/src/novatel_gps.cpp:947:36: error: no matching function for call to ‘toMsg(tf2::Quaternion&)’
  947 |       imu->orientation = tf2::toMsg(q);
```

```sh
novatel_gps_driver/novatel_gps_driver/src/nodes/novatel_gps_node.cpp:1146:10: fatal error: rclcpp_components/register_node_macro.hpp: No such file or directory
[build]  1146 | #include <rclcpp_components/register_node_macro.hpp>
```

I have fixed them and will open a PR.

--- a/novatel_gps_driver/CMakeLists.txt
+++ b/novatel_gps_driver/CMakeLists.txt
@@ -74,6 +74,7 @@ ament_target_dependencies(${PROJECT_NAME}
   nav_msgs
   novatel_gps_msgs
   rclcpp
+  rclcpp_components
   sensor_msgs
   std_msgs
   swri_math_util

--- a/novatel_gps_driver/CMakeLists.txt
+++ b/novatel_gps_driver/CMakeLists.txt
@@ -67,6 +67,16 @@ target_link_libraries(${PROJECT_NAME}
   Boost::boost
   ${libpcap_LIBRARIES}
 )
+
+# TODO: Remove once support is dropped
+if(tf2_geometry_msgs_VERSION VERSION_GREATER_EQUAL "0.18.0")
+  set(USE_TF2_GEOMETRY_MSGS_HPP USE_TF2_GEOMETRY_MSGS_HPP)
+endif()
+
+target_compile_definitions(${PROJECT_NAME} PRIVATE
+  ${USE_TF2_GEOMETRY_MSGS_HPP}
+)
+
 ament_target_dependencies(${PROJECT_NAME}
   diagnostic_msgs
   diagnostic_updater

--- a/novatel_gps_driver/src/novatel_gps.cpp
+++ b/novatel_gps_driver/src/novatel_gps.cpp
@@ -40,7 +40,7 @@
 #include <rclcpp/node.hpp>
 
 // TODO: Remove once support is dropped.
-#if __has_include(<tf2_geometry_msgs/tf2_geometry_msgs.hpp>)
+#ifdef USE_TF2_GEOMETRY_MSGS_HPP
 #  include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
 #else
 #  include <tf2_geometry_msgs/tf2_geometry_msgs.h>

--- a/novatel_gps_driver/src/novatel_gps.cpp
+++ b/novatel_gps_driver/src/novatel_gps.cpp
@@ -39,7 +39,12 @@
 #include <boost/make_shared.hpp>
 #include <rclcpp/node.hpp>
 
-#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+// TODO: Remove once support is dropped.
+#if __has_include(<tf2_geometry_msgs/tf2_geometry_msgs.hpp>)
+#  include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+#else
+#  include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#endif
 #include <tf2/LinearMath/Quaternion.h>
 #include <tf2/utils.h>
 

--- a/novatel_gps_driver/src/novatel_gps.cpp
+++ b/novatel_gps_driver/src/novatel_gps.cpp
@@ -39,6 +39,7 @@
 #include <boost/make_shared.hpp>
 #include <rclcpp/node.hpp>
 
+#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
 #include <tf2/LinearMath/Quaternion.h>
 #include <tf2/utils.h>
 

